File: 프로그래머스/1/42840. 모의고사/모의고사.java
class Solution {
    public int[] solution(int[] answers) {
        int[] answer = {};
        int[] result = new int[3];
        int max = 0;
        int len = 0;

        int[] one = {1,2,3,4,5};
        int[] two = {2,1,2,3,2,4,2,5};
        int[] three = {3,3,1,1,2,2,4,4,5,5};

        for(int i = 0; i < answers.length; i++){
            if(one[i % 5] == answers[i]) result[0]++;
            if(two[i % 8] == answers[i]) result[1]++;
            if(three[i % 10] == answers[i]) result[2]++;
        }

        max = Math.max(result[0], Math.max(result[1], result[2]));
        
        if(result[0] == max) len++;
        if(result[1] == max) len++;
        if(result[2] == max) len++;

        answer = new int[len];

        if(result[2] == max) answer[--len] = 3;
        if(result[1] == max) answer[--len] = 2;
        if(result[0] == max) answer[--len] = 1;
        
        return answer;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    
    a = [1, 2, 3, 4, 5]
    b = [2, 1, 2, 3, 2, 4, 2, 5]
    c = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    p1, p2, p3, pm = 0, 0, 0, 0
    
    for i, n in enumerate(answers):
        if a[i % 5] == n: p1 += 1
        if b[i % 8] == n: p2 += 1
        if c[i % 10] == n: p3 += 1
        
    pm = max(p1,p2,p3)
    
    if p1 == pm: answer.append(1)
    if p2 == pm: answer.append(2)
    if p3 == pm: answer.append(3)
    
    return answer